### DISCUSSION TITLE
Difference in the rendering of Chinese characters between autocad and "ezdxf view"

### DISCUSSION THREAD

**Discussion by maslke** [mozman/ezdxf#1075] [Q&A]
Hello everyone, I have a question I'd like to ask.

Regarding a same dxf file, when viewed through AutoCAD software and the "ezdxf view" method, there is a noticeable difference in the rendering of Chinese characters.

Here is a simple demo dfx file. When viewed in AutoCAD software, the effect is as follows. It can be observed that the text does not exceed the outer rectangle.
<img width="911" alt="image" src="https://github.com/mozman/ezdxf/assets/6869592/938a786f-5f25-4c04-b9df-4297147900df">

When viewed using the "ezdxf view" method, the effect is as follows. The rendered text exceeds the rectangle.
<img width="1196" alt="image" src="https://github.com/mozman/ezdxf/assets/6869592/cb62efd6-2677-49a3-afe4-82a635d89848">

What is the cause of this discrepancy, and is there a way to control this behavior?

Thank you all for your suggestions.

[123.zip](https://github.com/mozman/ezdxf/files/15259638/123.zip)

**Comment by maslke** [comment]
I hava just found issue #733 and I found this:
> All text rendering in ezdxf is done by the Matplotlib package which does not render fonts the same way as AutoCAD or BricsCAD. That's a fact you have to live with.

I want to convert dxf file with Chinese characters to pdf file using drawing addon. Does this process also depends on Matplotlib?

```pyhthon
backend = pymupdf.PyMuPdfBackend()
Frontend(RenderContext(doc), backend, config=cfg).draw_layout(dxf_layout)
pdf_bytes = backend.get_pdf_bytes(page, settings=settings)
```

Is there a way to control this?

**Reply by mozman** [reply]
Text is rendered by `fontTools` since version v1.1.0, the result is now closer to AutoCAD than rendered by `Matplotlib` but will never be perfect.

**Reply by maslke** [reply]
I noticed another issue, which may be related to ODAFileConverter.
In ezdxf, create a text entity and set the alignment to MIDDLE_LEFT.
After generating the dxf, when converted to dwg file through ODAFileConverter, the alignment of this text will change to BOTTOM_LEFT.
